packages/xpetra/src/CrsMatrix/Xpetra_TpetraCrsMatrix_def.hpp

```diff
@@ -335,6 +336,30 @@ namespace Xpetra {
     template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
     void TpetraCrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::apply(const MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > &X, MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > &Y, Teuchos::ETransp mode, Scalar alpha, Scalar beta) const { XPETRA_MONITOR("TpetraCrsMatrix::apply"); mtx_->apply(toTpetra(X), toTpetra(Y), mode, alpha, beta); }
 
+    template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+    void TpetraCrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::apply(const MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > &X, MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > &Y, Teuchos::ETransp mode, Scalar alpha, Scalar beta, bool sumInterfaceValues, const RCP<Import<LocalOrdinal, GlobalOrdinal, Node> >& regionInterfaceImporter, const Teuchos::ArrayRCP<LocalOrdinal>& regionInterfaceLIDs ) const
+    {
+        RCP<const Map< LocalOrdinal, GlobalOrdinal, Node >> regionInterfaceMap = regionInterfaceImporter->getTargetMap();
+        XPETRA_MONITOR("TpetraCrsMatrix::apply(region)");
+        mtx_->localApply(toTpetra(X), toTpetra(Y), mode, alpha, beta);
+        if (sumInterfaceValues)
+        {
+          // Step 2: preform communication to propagate local interface
+          // values to all the processor that share interfaces.
+          RCP<MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node >> matvecInterfaceTmp = MultiVectorFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Build(regionInterfaceMap, 1);
```

Comment:
Out of curiosity: Can't you use the pre-defined short names here to improve readability?
```suggestion
          RCP<MultiVector> matvecInterfaceTmp = MultiVectorFactory::Build(regionInterfaceMap, 1);
```

Reply:
The short names don't appear to be defined yet for this file. Should I add the shortnames header?

Reply:
@pohm01 Well, I think you can just include existing Xpetra's short name headers, right?

Reply:
I am not sure I see where the change has been made but the idea is to include this header: `Xpetra_UseShortNamesScalar.hpp`

Reply:
I don't think, that these changes have been made yet. Can't see them either.

Reply:
I haven't made the changes for this yet. Where do you suggest I put shortnames header? The only place I have found that does not cause problems is inside the function definition here.